lib/rules/wrong-quotes-for-template-string.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * @fileoverview Warn when using template string syntax in regular strings
+ * @author Jeroen Engels
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "Warn when using template string syntax in regular strings",
+            category: "Possible Errors",
+            recommended: true
+        },
+
+        schema: []
+    },
+
+    create: function(context) {
+        var regex = /\$\{.*\}/;
+
+        return {
+            Literal: function(node) {
+                if (typeof node.value === "string" && regex.test(node.value)) {
+                    context.report(node, "Possible use of template string inside a regular string.");
```

Comment:
I think we can use property shorthands (except `get` and `set`. `{get, set}` will cause syntax error for Node 4).
But it might be modified when we enable `object-shorthand` rule (related in #6407).

(FWIW, I would not like arrow functions here.)